lib/block-supports/colors.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Colors block support flag.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Add CSS classes and inline styles for colors to the incoming attributes array.
+ * This will be applied to the block markup in the front-end.
+ *
+ * @param  array $attributes comprehensive list of attributes to be applied.
+ * @param  array $block_attributes block attributes.
+ * @param  array $block_type Block type.
+ * @return array Colors CSS classes and inline styles.
+ */
+function gutenberg_apply_colors_support( $attributes, $block_attributes, $block_type ) {
+	$has_text_colors_support       = $block_type->supports && array_key_exists( '__experimentalColor', $block_type->supports );
+	$has_background_colors_support = $has_text_colors_support;
+	$has_link_colors_support       = $has_text_colors_support && array_key_exists( 'linkColor', $block_type->supports['__experimentalColor'] );
+	$has_gradients_support         = $has_text_colors_support && array_key_exists( 'gradients', $block_type->supports['__experimentalColor'] );
```

Comment:
I think currently experimental color can be defined as `true` and as an alternative to the array (in the case we just want basic support without linkColor or gradients).  In this case `array_key_exists` will error (and is doing so in the tests in `class-block-supported-styles.php`.  This is where that global styles supports function came in handy previously.

It doesn't look like we have any blocks currently defining it this way, so if we don't want to support being able to say `"__experimentalColor": true` as short for `"__experimentalColor": { "gradients": false }` maybe this is not an issue and we just need to update the test?

Reply:
Good catch. I agree we should just update the test here.

Reply:
Should be fixed.